cogs/shareLink.py

```diff
@@ -0,0 +1,45 @@
+import logging
+import random
+
+import spotipy
+from discord.ext import commands
+from discord.ext.commands.context import Context
+from spotipy.oauth2 import SpotifyClientCredentials
+
+from main import ManChanBot
+
+from .commandbase import CommandBase
+
+
+class ShareLink(CommandBase):
+    def __init__(self, bot: ManChanBot):
+        super().__init__(bot)
+        self.sp = spotipy.Spotify(
+            auth_manager=SpotifyClientCredentials(
+                client_id=self.configs["CLIENT_ID"],
+                client_secret=self.configs["CLIENT_SECRET"],
+            )
+        )
+
+    @commands.command()
+    async def track(self, ctx: Context, *args: str):
+        join_words = " ".join(args).strip()
+        results = self.sp.search(q=join_words, limit=1, type="track")
+        await ctx.channel.send(
+            results["tracks"]["items"][0]["external_urls"]["spotify"]
+        )
+
+    @commands.command()
+    async def album(self, ctx: Context, *args: str):
+        join_words = " ".join(args)
+        results = self.sp.search(q=join_words, limit=1, type="album")
+        await ctx.channel.send(
+            results["albums"]["items"][0]["external_urls"]["spotify"]
+        )
+
+
+async def setup(bot: ManChanBot):
+    if ShareLink.is_enabled():
```

Comment:
This will always default to True. We want to make sure it can be disabled anytime (future work will allow auto reloading commands).

https://github.com/anime-crossing/man-chan/blob/e0528c1b9d24f5d86d3d5c48929255fa7ff0bb80/cogs/socialcredit.py#L119

Here is an example on logic of how to enable/disable the commands, while making sure all settings are correct

Reply:
should be fixed